src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -71,58 +68,149 @@ class F extends ValidationTest {
 
 export const g = makeTestGroup(F);
 
-g.test('mipLevelCount_validation_on_formats')
-  .params(poptions('format', kAllTextureFormats))
+g.test('zero_size')
+  .desc(
+    `Test texture creation with zero or nonzero size of
+   width, height, depthOrArrayLayers and mipLevelCount for every dimension, and representative formats.`
+  )
+  .params(
+    params()
+      .combine(poptions('dimension', ['1d', '2d', '3d'] as const))
+      .combine(
+        poptions('zeroArgument', [
+          'none',
+          'width',
+          'height',
+          'depthOrArrayLayers',
+          'mipLevelCount',
+        ] as const)
+      )
+      .combine(
+        poptions('format', [
+          'rgba8unorm',
+          'rgb10a2unorm',
+          'depth24plus-stencil8',
+          'bc1-rgba-unorm',
+        ] as const)
+      )
+  )
   .fn(async t => {
-    const format: GPUTextureFormat = t.params.format;
+    const { dimension, zeroArgument, format } = t.params;
+
+    const size = dimension === '1d' ? [31, 1, 1] : dimension === '2d' ? [31, 31, 1] : [31, 31, 31];
+    let mipLevelCount = 1;
+
+    switch (zeroArgument) {
+      case 'width':
+        size[0] = 0;
+        break;
+      case 'height':
+        size[1] = 0;
+        break;
+      case 'depthOrArrayLayers':
+        size[2] = 0;
+        break;
+      case 'mipLevelCount':
+        mipLevelCount = 0;
+        break;
+      default:
+        break;
+    }
 
     const descriptor = {
-      size: { width: 32, height: 32, depth: 1 },
-      mipLevelCount: 6,
+      size,
+      mipLevelCount,
+      dimension,
       format,
       usage: GPUTextureUsage.SAMPLED,
     };
 
-    // const extension: GPUExtensionName | undefined = kAllTextureFormatInfo[format].extension;
-    const extension: GPUExtensionName = kAllTextureFormatInfo[format].extension;
-    await t.selectDeviceOrSkipTestCase({ extensions: [extension] });
+    await t.selectDeviceOrSkipTestCase(kAllTextureFormatInfo[format].extension);
 
-    t.device.createTexture(descriptor);
+    const success = zeroArgument === 'none';
+    t.expectValidationError(() => {
+      t.device.createTexture(descriptor);
+    }, !success);
   });
 
-g.test('mipLevelCount_validation_on_dimensions')
+g.test('mipLevelCount,format')
+  .desc(
+    `Test texture creation with no mipmap chain, partial mipmap chain, full mipmap chain, out-of-bounds mipmap chain
+    for every format with different texture dimension types.`
+  )
+  .params(
+    params()
+      .combine(poptions('dimension', ['1d', '2d', '3d'] as const))
+      .combine(poptions('format', kAllTextureFormats))
+      .combine(poptions('mipLevelCount', [1, 3, 5, 6]))
```

Comment:
Won't this test combinations that aren't valid? I don't think all formats can be 1d, 3d, and have mipmapping. We'll need capability info about this somewhere (in both cts and spec).

The validation tests for those things would be separate I think, but we still need to skip the invalid combinations here somehow.

Reply:
Removed test for 1D/3D, and added a TODO.